helix-core/src/comment.rs

```diff
@@ -45,8 +45,8 @@ fn find_line_comment(
 
             // determine margin of 0 or 1 for uncommenting; if any comment token is not followed by a space,
             // a margin of 0 is used for all lines.
-            if matches!(line_slice.get_char(pos + token_len), Some(c) if c != ' ') {
-                margin = 0;
+            if !matches!(line_slice.get_char(pos + token_len), Some(c) if c == ' ') {
+                margin = 0
```

Comment:
```suggestion
                margin = 0;
```

looks like this semicolon got chopped off on accident